extra/CMakeLists.txt

```diff
@@ -1,63 +1,79 @@
-set(nmodl_makefile "nrnivmodl_core_makefile")
-set(nmodl_compile_scipt "nrnivmodl-core")
+# =============================================================================
+# Copyright (C) 2016-2019 Blue Brain Project
+#
+# See top-level LICENSE file for details.
+# =============================================================================
 
-get_directory_property(CORENRN_COMPILE_FLAGS COMPILE_DEFINITIONS)
+# =============================================================================
+# Prepare nrnivmodl-core script with correct compiler flags
+# =============================================================================
 
-if (CORENRN_COMPILE_FLAGS)
-    set(CORENRN_COMPILE_DEFS "-D${CORENRN_COMPILE_FLAGS}")
-    string(REPLACE ";" " -D" CORENRN_COMPILE_DEFS "${CORENRN_COMPILE_DEFS}")
+# extract the COMPILE_DEFINITIONS property from the directory.
+get_directory_property(CORENRN_COMPILE_FLAGS COMPILE_DEFINITIONS)
+if(CORENRN_COMPILE_FLAGS)
+  set(CORENRN_COMPILE_DEFS "-D${CORENRN_COMPILE_FLAGS}")
```

Comment:
I've seen this pattern already above.. I think this should be done differently. First convert to list, then iterate and prepend `-D`. This works, but it's a bit weird.

Reply:
yeah agree.